src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java

```diff
@@ -623,6 +629,12 @@ private void executeCrafting(final IEnergyGrid eg, final CraftingGridCache cc) {
 
                             for (final IAEItemStack out : details.getCondensedOutputs()) {
                                 this.postChange(out, this.machineSrc);
+                                providers.computeIfAbsent(out.hashCode(), k -> new ArrayList<DimensionalCoord>());
```

Comment:
Why are putting hashcodes into the hashmap instead of letting it do the hashing?

Reply:
Can you explain why you can't just put `out` in the hashmap? hashCodes can have collisions (two different items mapping to the same hash code), which can cause visual bugs

Reply:
use less size to storage nbt data

Reply:
never be two different items has same hash code, be cause `hashCode()` using AEItemStack `hashCode()`, you can check AEItemStack how to generate hashcode

Reply:
@eigenraven  sorry you are right, i changed my code, now it's using `IAEItemStack` 
thanks @eigenraven  @Glease